src/main/java/com/marklogic/client/ext/modulesloader/impl/PropertiesModuleManager.java

```diff
@@ -130,21 +131,38 @@ public void saveLastLoadedTimestamp(File file, Date date) {
     }
 
     /**
-     * Lower-casing avoids some annoying issues on Windows where sometimes you get "C:" at the start, and other times
-     * you get "c:". This of course will be a problem if you for some reason have modules with the same names but
-     * differing in some cases, but I'm not sure why anyone would do that.
      *
      * @param file
      * @return
      */
-    protected String buildKey(File file) {
-        String path = file.getAbsolutePath().toLowerCase();
-        final String key = host != null ? host + ":" + path : path;
-        if (logger.isDebugEnabled()) {
-        	logger.debug("Key for file " + file.getAbsolutePath() + ": " + key);
-        }
-        return key;
-    }
+	protected String buildKey(File file) {
+		String path = normalizeDriveLetter(file);
+		final String key = host != null ? host + ":" + path : path;
+		if (logger.isDebugEnabled()) {
+			logger.debug("Key for file " + file.getAbsolutePath() + ": " + key);
+		}
+		return key;
+	}
+
+	/**
+	 * Lower-casing avoids some annoying issues on Windows where sometimes you get "C:" at the start, and other times
+	 * you get "c:".
+	 *
+	 * @param file
+	 * @return
+	 */
+	protected String normalizeDriveLetter(File file) {
```

Comment:
Sorry for the delay on this - I tried writing a test for this, but on Mac/Linux, "getRoot" always equals "/". Can you add tests for this - i.e. in a PropertiesModuleManagerTest (you'll want to rebase on dev) - that verify this on Windows? I assume the tests would just pass in Mac/Linux?

I was also wondering if an interface could be extracted here with "getPath()" and "getRoot()" methods. A real impl would be backed by a File. A test impl would be a mock that returns whatever we want. That allows us to test lines 157 through 164. 

If you like the above idea, let me know, I'll toss it together and then merge this in.